internal/npm_install/npm_install.bzl

```diff
@@ -130,22 +130,31 @@ data attribute.
         default = True,
     ),
 })
+YARN_ENVIRONMENT = [
 
-def _create_build_files(repository_ctx, rule_type, node, lock_file):
+]
+def _process_pnp_file(repository_ctx, node,workspace_path):
+    repository_ctx.report_progress("Processing PNP file: Patching module directory")
+    result = repository_ctx.execute([
+        node,
+        "process_pnp_file.js",
+       workspace_path
+    ])
+    if result.return_code:
+        fail("generate_build_file.js failed: \nSTDOUT:\n%s\nSTDERR:\n%s" % (result.stdout, result.stderr))
+
+def _create_build_files(repository_ctx, rule_type, node,workspace_path):
     error_on_build_files = repository_ctx.attr.symlink_node_modules and not repository_ctx.attr.always_hide_bazel_files
 
     repository_ctx.report_progress("Processing node_modules: installing Bazel packages and generating BUILD files")
     if repository_ctx.attr.manual_build_file_contents:
         repository_ctx.file("manual_build_file_contents", repository_ctx.attr.manual_build_file_contents)
     result = repository_ctx.execute([
         node,
-        "generate_build_file.js",
-        repository_ctx.attr.name,
-        rule_type,
-        "1" if error_on_build_files else "0",
-        repository_ctx.path(lock_file),
-        ",".join(repository_ctx.attr.included_files),
-    ])
```

Comment:
This should be changed to execute the new generator when using pnp, but still running the old one when not.